src/server/controllers/users.controller.js

```diff
@@ -196,3 +201,212 @@ function crawlByAccessTokenFacebook(accessToken) {
   });
 
 }
+
+export function getJobCategory(req, res, next) {
+  if (req.session._id) {
+    res.status(200).json(jobcategory);
+  } else {
+    res.status(401).json({ err_point: userCallback.ERR_FAIL_AUTH });
+  }
+}
+
+export function editGeneralProfile(req, res, next) {
+  if (req.session._id) {
+    let file = null;
+    let field = null;
+    let editData = null;
+    let form = new formidable.IncomingForm();
+    form.encoding = 'utf-8';
+    new Promise((resolve, reject) => {
+      form.parse(req, function (err, fields, files) {
+        if (err) {
+          reject();
+        } else {
+          file = files.image; // file when postman test.
+          field = JSON.parse(unescape(fields.info));
+          editData = {
+            name: field.name,
+            languages: field.languages,
+            location: field.location,
+            about: field.about,
+            education: field.education,
+            work: field.work,
+          };
+          resolve(true);
+        }
+      });
+    })
+    .then(parsed => {
+      if (parsed) {
+        return User.findOne({ _id: req.session._id, email: { $ne: null } }).exec();
+      } else {
+        throw new err;
+      }
+    })
+    .then(userWithEmail => {
+      if (!userWithEmail) {
+        if (field.email === null || field.email === undefined) {
```

Comment:
Well, not specific one because I don't know how the client request.
but I think client would use one of that ways.
Send null, or not send.